synapse/http/federation/srv_resolver.py

```diff
@@ -103,11 +103,16 @@ class SrvResolver:
 
     Args:
         dns_client (twisted.internet.interfaces.IResolver): twisted resolver impl
-        cache (dict): cache object
-        get_time (callable): clock implementation. Should return seconds since the epoch
+        cache: cache object
+        get_time: clock implementation. Should return seconds since the epoch
     """
 
-    def __init__(self, dns_client=client, cache=SERVER_CACHE, get_time=time.time):
+    def __init__(
+        self,
+        dns_client=client,
```

Comment:
You'll probably need to import this.

```suggestion
        dns_client: IResolver = client,
```

(Also please remove it from the docstring.)

Reply:
I tried this and got new errors. I will reproduce it an post it here.

Reply:
With `dns_client: IResolver = client,`

I get this:
```
synapse/http/federation/srv_resolver.py:113: error: Incompatible default for argument "dns_client" (default has type Module, argument has type "IResolver")  [assignment]
synapse/http/federation/srv_resolver.py:143: error: Missing positional argument "timeout" in call to "lookupService" of "IResolver"  [call-arg]
synapse/http/federation/srv_resolver.py:143: error: Argument 1 to "lookupService" of "IResolver" has incompatible type "bytes"; expected "str"  [arg-type]
```

Reply:
That's unfortunate, it seems that some of the Twisted types around DNS are odd. You can leave it for now and we can take another look in the future.